src/runtime_src/core/edge/user/aie/AIEResources.h

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2020, Xilinx Inc - All rights reserved
+ * ZNYQ XRT Library layered on top of ZYNQ zocl kernel driver
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <stddef.h>
+#include <vector>
+
+extern "C"
+{
+#include "xaiengine/xaiegbl.h"
+}
+
+#define NUM_PERF_COUNTERS_PL 2
+#define NUM_PERF_COUNTERS_MEM 2
+#define NUM_PERF_COUNTERS_CORE 4
+#define NUM_TRACE_EVENTS 8
+#define NUM_STREAM_SWITCH_EVENT_PORTS 8
+#define NUM_EVENT_BROADCASTS 16
+#define NUM_PROGRAM_COUNTERS 4
+#define CORE_MODULE_TYPE 0
+#define MEMORY_MODULE_TYPE 1
+#define PL_MODULE_TYPE 2
```

Comment:
These #defines in the header is not the greatest.  Anyone including this header automatically sees the defines the preprocessor will do its magic.   Prefer enumeration or constexpr.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Renum-macro

At a minimum at least prefix the defines to uniquify them.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Res-MACROS